### PR TITLE
Added support of MI public endpoint in Login Migration

### PR DESCRIPTION
**Pull request by Bhavikshah123**
- This PR adds the support for public endpoint for SQL Managed instance in login migration by providing a target port field along with the existing target username and password. The port was defaulted to 1433 in the backend previously which is a private endpoint default port. By provided a field to enter the port, we are adding the flexibility to enter the user desired port. The port 3342, the public endpoint default port. can also be entered now. 

image

image

- Added Port validation - the port number is a required field and the value entered should be a valid number in the range of 0-65535. The connect button gets enabled on the valid port value along with existing checks.

- Added logic to generate MI target server name based on the endpoint selected (public or private)
  Public endpoint format: <mi_name>.public.<dns_zone>.database.windows.net
  Private endpoint format: <mi_name>.<dns_zone>.database.windows.net

Testing on SQL migration extension on ADS
- Verified successful validation checks for target Port field.
- Verified successful connection to Public and Private MI endpoint.
- Verified successful fetch of target login for Public and Private MI endpoint.
- Verified the successful creation of connection string for Public and Private MI endpoint.
- Verified the successful login migrations for MI with private and public endpoint.

### CODE REVIEW

**Review by stuti149 — APPROVED** [review]

File: extensions/sql-migration/src/wizard/loginMigrationTargetSelectionPage.ts [hunk 1]
```diff
@@ -37,12 +37,18 @@ export class LoginMigrationTargetSelectionPage extends MigrationWizardPage {
...
 	private _resourceAuthenticationContainer!: azdata.FlexContainer;
+	private _targetPortLabel!: azdata.TextComponent;
 	private _targetUserNameInputBox!: azdata.InputBoxComponent;
 	private _targetPasswordInputBox!: azdata.InputBoxComponent;
+	private _targetPortInputBox!: azdata.InputBoxComponent;
 	private _testConectionButton!: azdata.ButtonComponent;
 	private _connectionResultsInfoBox!: azdata.InfoBoxComponent;
 	private _migrationTargetPlatform!: MigrationTargetType;
 
+	private readonly _defaultSqlPort: string = "1433";
```
Comment (brian-harris) [inline extensions/sql-migration/src/wizard/loginMigrationTargetSelectionPage.ts:48]: should the default port for SQL MI be used here:  3342

File: extensions/sql-migration/src/wizard/loginMigrationTargetSelectionPage.ts [hunk 2]
```diff
@@ -863,23 +877,84 @@ export class LoginMigrationTargetSelectionPage extends MigrationWizardPage {
...
+	private _validatePort(port: string): void {
+		// check if port is empty
+		if (port === "") {
+			this._isPortValid = false;
+			return;
+		}
+
+		// Check if the port is a number, port is an integer and port is in the range of 0 to 65535
+		let portNumber = Number(port);
+		if (isNaN(portNumber) || !Number.isInteger(portNumber) || portNumber < 0 || portNumber > 65535) {
```
Comment (brian-harris) [inline extensions/sql-migration/src/wizard/loginMigrationTargetSelectionPage.ts:941]: you might be able to narrow you valid ranges down more I use this range for Oracle
export const OraclePortMinimum = 1024;
export const OraclePortMaximum = 65536;

**Review by brian-harris — APPROVED** [review]

File: extensions/sql-migration/src/wizard/loginMigrationTargetSelectionPage.ts (hunk 1 above)
Comment (Bhavikshah123) [reply extensions/sql-migration/src/wizard/loginMigrationTargetSelectionPage.ts:48]: SQL MI by default has private endpoint enabled with port 1433.  3342 is the default port for public endpoint. So keeping 1433 as the default port.

File: extensions/sql-migration/src/wizard/loginMigrationTargetSelectionPage.ts (hunk 2 above)
Comment (Bhavikshah123) [reply extensions/sql-migration/src/wizard/loginMigrationTargetSelectionPage.ts:941]: Keeping it 0 to 65535 for now, considering we want to provide generic range, and it should be able to take care of future changes as suggested by PM's.

**Pull request merged**